### ISSUE TITLE
关于数据集设置的问题

### ISSUE THREAD

**Issue by xbd1519722757** [tsinghua-fib-lab/UV-SAM#3]
作者您好！请问在跑这个代码之前，是否要自行将数据集中每张label的值修改成0、1？如果我想设置inference得到的结果颜色，请问是不是应该修改UV-SAM-main\mmpl\datasets\uv_seg_dataset.py中的以下代码？
```
METAINFO = {
        'classes': ('background', 'UV',),
        'palette': [[120, 120, 120], [180, 120, 120],]
    }
```
另外，在您公布的数据集中，标签图像background和UV的值似乎分别是[0,0,0]和[128,0,0]，为什么上述代码要设置为[120, 120, 120]和[180, 120, 120]？
真诚盼望得到您的解答。祝您元宵节快乐！

**Comment by xbd1519722757** [comment]
另外想请教一下，如果想用这个模型跑多分类数据，除了修改train.py、SegSAMPLerEmbAdd_config.py，并在mmpl/datasets里添加自己的数据集信息外，还应该修改哪些文件？现在loss和IOU算出来似乎不太正常，有点伤脑筋。

**Comment by NYB-AAA** [comment]
请问你解决问题了吗，我在源码上训练作者给出的数据集，UV部分的iou很低，不知道是什么原因

**Comment by xbd1519722757** [comment]
还没有解决

**Comment by NYB-AAA** [comment]
你用作者原文的数据集跑出来的loss曲线正常吗，为什么我这个一直在波动，这是我第一次训练得到的数据，用的预训练权重是segformer的mit_b0.pth，北京数据集，
lr-AdamW/pg1:0.000004933178929321101
lr-AdamW/pg2:0.000004933178929321101
lr-AdamW-1/pg1:0.000004933178929321101
lr-AdamW-1/pg2:0.000004933178929321101
trainer/global_step:8599
_timestamp:1710872500.6427822
_runtime:37462.0598192215
_step:959
train_loss:0.6227031350135803
train_loss_ce:0.3675955533981323
train_acc_seg:82.71102905273438
train_loss_SAM:0.255107581615448
loss:0.6227031350135803
max_grad:0.47632843255996704
min_grad:0.0009266319102607667
epoch:99
valaacc_0:0.8882473707199097
valmiou_0:65.59
valmacc_0:73.48
valiou_0:0.877764105796814
valiou_1:0.4340989887714386
valacc_0:0.9679312705993652
valacc_1:0.5016538500785828

**Comment by NYB-AAA** [comment]
而且测试集上的结果也和论文里差了将近十个点，全是论文里的参数和预训练权重，你当时测试集上结果咋样啊
![image](https://github.com/tsinghua-fib-lab/UV-SAM/assets/130569364/11df41cf-5c02-441e-bbfb-ff261bd4232e)

**Comment by xbd1519722757** [comment]
不好意思，我还没完整跑完过。当时我跑了一会儿发现iou有点怪，又因为是租服务器跑的，考虑到成本问题，就先搁置了，想等作者解答了再跑。

**Comment by NYB-AAA** [comment]
头疼，复现不出来啊，不知道哪里出了问题，我也是租服务器跑的，也不清楚作者用的什么卡，想控制变量都不行，又是emo的一天



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年3月20日(周三) 下午4:28
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [tsinghua-fib-lab/UV-SAM] 关于数据集设置的问题 (Issue #3)




 
不好意思，我还没完整跑完过。当时我跑了一会儿发现iou有点怪，又因为是租服务器跑的，考虑到成本问题，就先搁置了，想等作者解答了再跑。

**Comment by Xin-Zh00** [comment]
I used PIL's Image library to process images. Please refer to inference_seg.py.
As for the palette and transferring to other datasets, please refer to mmseg codes.

> 另外想请教一下，如果想用这个模型跑多分类数据，除了修改train.py、SegSAMPLerEmbAdd_config.py，并在mmpl/datasets里添加自己的数据集信息外，还应该修改哪些文件？现在loss和IOU算出来似乎不太正常，有点伤脑筋。

**Comment by Xin-Zh00** [comment]
> 而且测试集上的结果也和论文里差了将近十个点，全是论文里的参数和预训练权重，你当时测试集上结果咋样啊 ![image](https://private-user-images.githubusercontent.com/130569364/314423924-11df41cf-5c02-441e-bbfb-ff261bd4232e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4NDk4MTAsIm5iZiI6MTcxMjg0OTUxMCwicGF0aCI6Ii8xMzA1NjkzNjQvMzE0NDIzOTI0LTExZGY0MWNmLTVjMDItNDQxZS1iYmZiLWZmMjYxYmQ0MjMyZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMVQxNTMxNTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNzkxMTBkZWI1YjAxZGM5ZTQzMTJjYjI2NjY0OGU0YjZlY2EyMmU3M2RiNDA3ZWRmMDBlZTExZTQ4OWUwMWI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.auAY-NvEPkiwsvFkA9lN79TQQTfbViOHk9zjLAlUlFo)

Hi, I am sorry that I did not provide the step-by-step guidance for the implementation of the code. I have updated the README.md file to provide more detailed guidance. If you have any questions, please feel free to contact me. Thank you for your understanding.

**Comment by Xin-Zh00** [comment]
> NYB



> 你用作者原文的数据集跑出来的loss曲线正常吗，为什么我这个一直在波动，这是我第一次训练得到的数据，用的预训练权重是segformer的mit_b0.pth，北京数据集， lr-AdamW/pg1:0.000004933178929321101 lr-AdamW/pg2:0.000004933178929321101 lr-AdamW-1/pg1:0.000004933178929321101 lr-AdamW-1/pg2:0.000004933178929321101 trainer/global_step:8599 _timestamp:1710872500.6427822 _runtime:37462.0598192215 _step:959 train_loss:0.6227031350135803 train_loss_ce:0.3675955533981323 train_acc_seg:82.71102905273438 train_loss_SAM:0.255107581615448 loss:0.6227031350135803 max_grad:0.47632843255996704 min_grad:0.0009266319102607667 epoch:99 valaacc_0:0.8882473707199097 valmiou_0:65.59 valmacc_0:73.48 valiou_0:0.877764105796814 valiou_1:0.4340989887714386 valacc_0:0.9679312705993652 valacc_1:0.5016538500785828

It seems that your pre-trained model is not loaded correctly. You can try to use the pre-trained model provided by us to see if the loss curve is normal. If the loss curve is still fluctuating, you can try to adjust the learning rate or the optimizer to see if it can be improved. Our loss curve could be found in the **./assets/** path and the training log could be found in the **./log/** path.

**Comment by Xin-Zh00** [comment]
> 另外想请教一下，如果想用这个模型跑多分类数据，除了修改train.py、SegSAMPLerEmbAdd_config.py，并在mmpl/datasets里添加自己的数据集信息外，还应该修改哪些文件？现在loss和IOU算出来似乎不太正常，有点伤脑筋。

I have not tried to use this model for multi-classification tasks, but I think you can try to modify the following files: train.py, SegSAMPLerEmbAdd_config.py, and add your dataset information in mmpl/datasets.
